travis/docker.sh

```diff
@@ -0,0 +1,51 @@
+#!/bin/sh
+
+if [ "${DOCKER_IMAGE_NAME}" = "" ]
+then
+  echo "Docker deployment error. Env DOCKER_IMAGE_NAME has not been provided"
+  exit 128
+fi
+
+if [ "${DOCKER_USERNAME}" = "" ]
+then
+  echo "Docker deployment error. Env DOCKER_USERNAME has not been provided"
+  exit 128
+fi
+
+if [ "${DOCKER_PASSWORD}" = "" ]
+then
+  echo "Docker deployment error. Env DOCKER_PASSWORD has not been provided"
+  exit 128
+fi
+
+echo "Login into docker..."
+echo "${DOCKER_PASSWORD}" | docker login -u "${DOCKER_USERNAME}" --password-stdin
+
+CURRENT_VERSION=$(npm run version --silent)
+
+echo "Creating image ${DOCKER_IMAGE_NAME}:${CURRENT_VERSION}"
+docker build -t "${DOCKER_IMAGE_NAME}:${CURRENT_VERSION}" .
+
+if [ "$TRAVIS_BRANCH" = "$MAIN_BRANCH" ]
+then
+    echo "Building for ${$MAIN_BRANCH} branch..."
+
+    echo "Docker tagging alpha version"
+    docker tag "${DOCKER_IMAGE_NAME}:${CURRENT_VERSION}" "${DOCKER_IMAGE_NAME}:${CURRENT_VERSION}-alpha"
+    docker tag "${DOCKER_IMAGE_NAME}:${CURRENT_VERSION}" "${DOCKER_IMAGE_NAME}:${CURRENT_VERSION}-alpha-$(date +%Y%m%d%H%M)"
+
+    echo "Docker pushing alpha"
+    docker push "${DOCKER_IMAGE_NAME}:${CURRENT_VERSION}-alpha"
+    docker push "${DOCKER_IMAGE_NAME}:${CURRENT_VERSION}-alpha-$(date +%Y%m%d%H%M)"
+fi
+
+if [ "$TRAVIS_BRANCH" = "$RELEASE_BRANCH" ]
+then
+    echo "Building for ${$RELEASE_BRANCH} branch..."
+    echo "Docker tagging release version"
+    docker tag "${DOCKER_IMAGE_NAME}:${CURRENT_VERSION}" "${DOCKER_IMAGE_NAME}:release"
+
+    echo "Docker pushing release"
+    docker push "${DOCKER_IMAGE_NAME}:release"
+    docker push "${DOCKER_IMAGE_NAME}:${CURRENT_VERSION}"
```

Comment:
Automatic steps you can do in the release branch:

* tag
* update patch version
* push the version update

Have a look at the release.sh we are using in many projects using the "release" branch trigger process.

https://github.com/nemtech/symbol-sdk-typescript-javascript/blob/main/travis/release.sh

Reply:
Yeah, we have it https://github.com/nemgrouplimited/symbol-explorer/blob/main/travis/release.sh

But i seperate it, because I put everything releated to Docker in one file.

Reply:
I've missed that in the Travis file. 

You may want to swap the steps in Travis YAML.  First, do the docker image release push, and then call release.sh that tags and increments the version.